sound/soc/sof/ipc4-topology.c

```diff
@@ -541,7 +541,23 @@ static int sof_ipc4_widget_setup_comp_dai(struct snd_sof_widget *swidget)
 	{
 		struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(scomp);
 		struct sof_ipc4_alh_configuration_blob *blob;
+		struct snd_soc_dapm_path *p;
 		struct snd_sof_widget *w;
+		int src_num = 0;
+
+		snd_soc_dapm_widget_for_each_source_path(swidget->widget, p)
+			src_num++;
+
+		if (swidget->id == snd_soc_dapm_dai_in && src_num == 0) {
+			/*
+			 * The blob will not be used if the ALH copier is playback direction
```

Comment:
I don't see any indication in the patch that this is only applicable for ALH copier. It might be that you saw this with ALH, but nothing is locking this to ALH and if there is, then should not we check if the copier is ALH?

Reply:
> I don't see any indication in the patch that this is only applicable for ALH copier. It might be that you saw it with ALH, but nothing is locking this to ALH and if there is, then should not we check if the copier is ALH?

@ujfalusi The code is inside the ipc4_copier->dai_type = SOF_DAI_INTEL_ALH switch case. So, the copier must be an ALH copier.